### PR TITLE
[docs] Fix typoes in comment in source code

### PR DESCRIPTION
**Pull request by aeroastro**
While reading some code, I have found out some typoes.

* Gramatical Typoes
* Comment on query_and_fetch, which has already been removed. (#22996)

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/search/SearchService.java [hunk 1]
```diff
@@ -963,7 +963,7 @@ public boolean canMatch(ShardSearchRequest request) throws IOException {
     }
 
     /**
-     * Returns true iff the given search source builder can be early terminated by rewriting to a match none query. Or in other words
+     * Returns true if the given search source builder can be early terminated by rewriting to a match none query. Or in other words
```
Comment (cbuescher) [inline core/src/main/java/org/elasticsearch/search/SearchService.java]: I think `iff` is intended here, meaning "if and only if" here (https://en.wikipedia.org/wiki/If_and_only_if), I'd leave this as is.

**Review by cbuescher — CHANGES_REQUESTED** [review]
@aeroastro thanks for the changes, I agree with all of them but one where I left a comment. Can you update, then I will merge this.

**Comment by aeroastro** [comment]
Thank you for your review.
I have fixed the issue by leaving iff as is at 1cfcda4c2cb26a053f7a3bace0b04dff48795022

File: core/src/main/java/org/elasticsearch/search/SearchService.java [hunk 2]
```diff
@@ -963,8 +963,8 @@ public boolean canMatch(ShardSearchRequest request) throws IOException {
     }
 
     /**
-     * Returns true iff the given search source builder can be early terminated by rewriting to a match none query. Or in other words
-     * if the execution of a the search request can be early terminated without executing it. This is for instance not possible if
```
Comment (cbuescher) [inline core/src/main/java/org/elasticsearch/search/SearchService.java]: Looks like you just changed both the first occurance of "iff" and the later "if" to the other version. Could you just change them back?

**Review by cbuescher — CHANGES_REQUESTED** [review]
@aeroastro thanks, looks like instead of changing back you changed another occurance of "if". Left a comment, can you correct this please?

**Comment by aeroastro** [comment]
Sorry about that.
I have fixed the mistake at e4123c9

**Review by cbuescher — APPROVED** [review]
Thanks, LGTM. I will merge this soon.

**Comment by aeroastro** [comment]
Thank you 🐱

**Pull request merged**